chat-commands.js

```diff
@@ -2316,7 +2316,14 @@ exports.commands = {
 			return this.errorReply(`[${duplicates.join(', ')}] ${Chat.plural(duplicates, "are", "is")} already blacklisted.`);
 		}
 
+		let userIndex = Config.groupsranking.indexOf(user.getIdentity(room).charAt(0));
 		for (const userid of targets) {
+			let targetUserAuthIndex = room.auth[userid] ? Config.groupsranking.indexOf(room.auth[userid]) : 0;
+			const targetUserGlobalAuthIndex = Users.usergroups[userid] ? Config.groupsranking.indexOf(Users.usergroups[userid].charAt(0)) : 0;
+			// use highest auth, may it be room or global
```

Comment:
That is way overengineering...

Reply:
?

Reply:
I updated this bit but the logic is kind of the same...